From: Andrea Righi <andrea.righi@canonical.com>
Subject: check for power state only when ACPI is available
Description: In bbswitch_off() we use acpi_bus_get_device() to determine the power
 state, but this feature is only available if ACPI is enabled in the kernel, so
 make sure to do this check only when ACPI is available to prevent build
 errors.
Bug: https://github.com/Bumblebee-Project/bbswitch/pull/216
---
 bbswitch.c | 2 ++
 1 file changed, 2 insertions(+)

Index: bbswitch-0.8/bbswitch.c
===================================================================
--- bbswitch-0.8.orig/bbswitch.c
+++ bbswitch-0.8/bbswitch.c
@@ -264,6 +264,7 @@ static void bbswitch_off(void) {
     pci_save_state(dis_dev);
     pci_clear_master(dis_dev);
     pci_disable_device(dis_dev);
+#ifdef CONFIG_ACPI
     do {
         struct acpi_device *ad = NULL;
         int r;
@@ -278,6 +279,7 @@ static void bbswitch_off(void) {
             ad->power.state = ACPI_STATE_D0;
         }
     } while (0);
+#endif
     pci_set_power_state(dis_dev, PCI_D3cold);
 
     if (bbswitch_acpi_off())
